tools/scenario-player/scenario_player/utils.py

```diff
@@ -0,0 +1,179 @@
+import time
+import uuid
+from collections import deque
+from datetime import datetime
+from itertools import islice
+from typing import Union
+
+import requests
+import structlog
+from eth_utils import to_checksum_address
+from requests.adapters import HTTPAdapter
+from web3.gas_strategies.time_based import fast_gas_price_strategy, medium_gas_price_strategy
+
+from raiden.network.rpc.client import JSONRPCClient, check_address_has_code
+from raiden.network.rpc.smartcontract_proxy import ContractProxy
+from raiden_contracts.constants import CONTRACT_CUSTOM_TOKEN
+from raiden_contracts.contract_manager import CONTRACT_MANAGER
+from scenario_player.exceptions import ScenarioTxError
+
+log = structlog.get_logger(__name__)
+
+
+# Seriously requests? For Humans?
+class TimeOutHTTPAdapter(HTTPAdapter):
+    def __init__(self, *args, **kwargs):
+        self.timeout = kwargs.pop('timeout', None)
+        super().__init__(*args, **kwargs)
+
+    def send(self, *args, **kwargs):
+        if 'timeout' not in kwargs or not kwargs['timeout']:
+            kwargs['timeout'] = self.timeout
+        return super().send(*args, **kwargs)
+
+
+class FrozenDict(dict):
```

Comment:
We don't seem to need this as it's not used in this PR.

Reply:
Right, that's also an old leftover. Thanks for noticing :)